src/libAtomVM/nifs.c

```diff
@@ -4242,6 +4248,36 @@ static term nif_code_load_binary(Context *ctx, int argc, term argv[])
     return result;
 }
 
+static term nif_lists_reverse(Context *ctx, int argc, term argv[])
+{
+    // Compared to erlang version, compute the length of the list and allocate
+    // at once the space for the reverse.
+    int proper;
+    size_t len = term_list_length(argv[0], &proper);
+    if (!proper) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    if (UNLIKELY(memory_ensure_free_with_roots(ctx, len * CONS_SIZE, 2, argv, MEMORY_CAN_SHRINK) != MEMORY_GC_OK)) {
+        RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+    }
+
+    term result = term_nil();
+    if (argc == 2) {
+        result = argv[1];
+    }
+    term list_crsr = argv[0];
+    while (!term_is_nil(list_crsr)) {
+        if (UNLIKELY(!term_is_nonempty_list(list_crsr))) {
+            RAISE_ERROR(BADARG_ATOM);
```

Comment:
This cannot happen, improper lists are ruled out before this loop.
So let's add a comment for extra clarity.